theme/gatsby-node.js

```diff
@@ -74,6 +75,38 @@ exports.createPages = async ({graphql, actions}, themeOptions) => {
   )
 }
 
+exports.onPostBuild = async ({graphql}) => {
+  try {
+    const {data} = await graphql(`
+      query {
+        allSitePage(filter: {context: {frontmatter: {componentId: {ne: null}}}}) {
+          nodes {
+            path
+            context {
+              frontmatter {
+                componentId
+                status
+              }
+            }
+          }
+        }
+      }
+    `)
+
+    const components = data.allSitePage.nodes.map(node => {
+      return {
+        id: node.context.frontmatter.componentId,
+        path: node.path,
+        status: node.context.frontmatter.status.toLowerCase()
+      }
+    })
+
+    fs.writeFileSync(path.resolve(process.cwd(), 'public/components.json'), JSON.stringify(components))
```

Comment:
I'm seeing the `Unable to build components.json` warning in both of the Vercel builds 😞 maybe it's the `process.cwd()` working differently between local and prod? [This article](https://www.seancdavis.com/blog/programmatically-create-json-pages-gatsby/) seems to say that `writeFileSync('./public/components.json'...` should work but I don't know how that's different to this. Maybe try `path.resolve(__dirname, 'public/components.json'`? Tbh I've never understood how absolute/relative paths in node scripts are treated, I usually brute force until I get a solution 😬 
It's definitely happening post build but maybe logging out what `readdirSync` returns would help figure it out?

Other than that the code all looks neat!

Reply:
Oh, that's expected here since none of the files in this site have `componentId` frontmatter. I just added `componentId` to a couple markdown files so we can use those as a test 👍

Reply:
Worked 🎉  https://doctocat-c8q7pu680-primer.vercel.app/doctocat/components.json

Reply:
Ah cool cool. I would have thought it would make a JSON file of just `[]` though? Anyway, it looks all good 👍